plugins/defaultplugins/rpc/notifications.go

```diff
@@ -43,10 +43,31 @@ func (svc *NotificationSvc) Get(from *rpc.NotificationRequest, server rpc.Notifi
 	svc.mx.RLock()
 	defer svc.mx.RUnlock()
 
+	// copy requested index locally
```

Comment:
@VladoLavor this is my proposal to simplify the implementation for getting the notifications, it also fixes the failing test case when requesting all notifications and buffer is exactly full. Please check it out.